core/utils.c

```diff
@@ -948,6 +948,11 @@ void
 mutex_delete(mutex_t *lock)
 {
     LOG(GLOBAL, LOG_THREADS, 3, "mutex_delete lock "PFX"\n", lock);
+    /* When doing detach, application locks may be held on threads which have
+     * already been interrupted and will never execute lock code again. Just
+     * ignore the assert on the lock_requests field below in those cases.
+     */
+    bool skip_lock_request_assert = false;
```

Comment:
(Some compiler is likely to complain about an unused var in release build)

Reply:
Done (in local repo currently)